### PR TITLE
Add annotate gem for Gemfile and model annotations

### LINKED ISSUE #1672: Install annotate gem for useful automatic documentation
The [annotate](https://github.com/ctran/annotate_models) gem automatically updates each model file with documentation showing the schema structure. This is very helpful for developers to refer to while they're getting to know the code.

The schema info is automagically updated with each database migration.

Annotate also provides annotations for the gems within the `Gemfile`, automatically adding a description and github URL for each gem.

#### Pros:
- Makes it much easier for new contributors to understand the code
- Easy to set up
- Well supported
- Well supported by documentation generation tools

#### Cons:
- Adds additional changes to commits when the database structure is changed
- Adds lots of documentation to the top of each model file

### PR DESCRIPTION
**Pull request by Bodacious**
Fixes #1672 .

Changes proposed in this PR:
- Added Annotate gem for helpful model annotations
- Added Annotate-Gem gem for helpful Gemfile annotations

### CODE REVIEW

**Comment by briri** [comment]
had a Travis bundler error. I just restarted the build on Travis. I've seen it resolve itself on those bundler errors in the past after a quick restart.

**Comment by briri** [comment]
@Bodacious seems to be a discrepency between the Gemfile and the Gemfile.lock. You added `protected_attributes` but it doesn't make an appearance in the lock file. I should be able to add it in with a bundle install on your end.

File: Gemfile [hunk 1]
```diff
@@ -3,81 +3,155 @@ source 'https://rubygems.org'
...
+
+# Tools for creating, working with, and running Rails applications. (http://www.rubyonrails.org)
 gem 'railties'
 
+# GEMS ADDED TO HELP HANDLE RAILS MIGRATION FROM 3.x to 4.2
+# THESE GEMS HELP SUPPORT DEPRACATED FUNCTIONALITY AND WILL LOSE SUPPORT IN
+# FUTURE VERSIONS WE SHOULD CONSIDER BRINGING THE CODE UP TO DATE INSTEAD
+
+# Protect attributes from mass assignment in Active Record models (https://github.com/rails/protected_attributes)
+gem 'protected_attributes', '~> 1.1.3'
```
Comment (xsrust) [inline Gemfile]: For some reason this wants to add back in protected_attributes which was removed by another PR.
Might be due to the merge-order

Comment (Bodacious) [reply Gemfile]: Yea, it's because development has updated. I'll rebase

**Pull request merged**